[ttzuche] iP

ttzuche opened the pull request:
# Bean

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Bean frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/nus-cs2103-AY2425S1/ip/pull/106).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉👍
 
And it is **FREE!**

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the ```main``` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

ttzuche committed:
Rename file, add greet and exit lines

ttzuche committed:
Add methods to follow OOP and logic for echo

ttzuche committed:
Add tasks to field, method to display list

ttzuche committed:
Add Task class

ttzuche committed:
Add mark and unmark methods, fix styling for printList in Bean.java

ttzuche committed:
Add Todo, Deadline and Event classes

ttzuche committed:
Modify toString methods in tasks sub-classes, display number of tasks

ttzuche committed:
Modify toString methods in tasks sub-classes, display number of tasks

ttzuche committed:
Create input.txt and EXPECCTED.TXT for auto testing, reformat toString output

ttzuche committed:
Add 2 Exception classes and apply to the chatbox program

ttzuche committed:
Add delete task function, add extra test for delete in input.txt

ttzuche committed:
Implement loadTasks and saveTasks methods in Bean.java

ttzuche committed:
Merge branch-Level-7

ttzuche committed:
Implement dates and times into deadline and event tasks

ttzuche committed:
Merge branch 'branch-Level-8' into master

ttzuche committed:
Add Command, Parser, Storage, TaskList, Ui for more oop

ttzuche committed:
Merge branch-moreoop into master

ttzuche committed:
Organise the classes to suitable packages

ttzuche committed:
Merge branch-package into master

ttzuche committed:
Add enableAssertions to be true

ttzuche committed:
Merge branch 'add-gradle-support'

ttzuche committed:
Fix build.gradle main file name

ttzuche committed:
Add junit tests for Parser, DeadlineTask and TodoTask

ttzuche committed:
Merge branch-junit into master

ttzuche committed:
Add javadocs for most classes

ttzuche committed:
Tweak code to comply with coding standard

ttzuche committed:
Implement find task function into application

ttzuche committed:
Fix merge conflict for branch-coding-standard

ttzuche committed:
Resolve merge conflict with branch-Level-9

ttzuche committed:
Implement checkstyle to detect coding style violations

ttzuche committed:
Merge branch-A-CodingStyle into master

ttzuche committed:
Add Main.java

ttzuche committed:
Create a working GUI without bean functionalities

ttzuche committed:
Implement Bean with its functionalities

ttzuche committed:
Update Javadocs for Ui, DialogBox, Launcher, Main and MainWindow

RadieonAjax commented on src/main/java/bean/Bean.java:
```diff
@@ -0,0 +1,143 @@
+     *
+     * @param args Command line arguments (not used).
+     */
+    public static void main(String[] args) {
+        new Bean("data/bean.txt").startConversation();
+    }
```
```suggestion
private static final String FILE_PATH = "./data/bean.txt"
    public static void main(String[] args) {
        new Bean("FILE_PATH").startConversation();
    }
```

I would suggest that you declare a constant variable for the file path as it makes it clear what was being passed as the parameter.

RadieonAjax commented on src/main/java/bean/parser/Parser.java:
```diff
@@ -0,0 +1,45 @@
+ */
+public class Parser {
+
+    /**
+     * Parses the user's input string and returns a Command object.
+     * If the command is not recognised, throws an UnknownCommandException.
+     *
+     * @param input The user's input string.
+     * @return A Command object corresponding to the user's input.
+     * @throws UnknownCommandException If the command is not recognised.
+     */
```
Great job for leaving comments about the parameters of the methods and the exceptions it throws in your header!

RadieonAjax commented on src/main/java/bean/ui/Ui.java:
```diff
@@ -0,0 +1,140 @@
+     * Displays a greeting message to the user.
+     */
+    public void showGreeting() {
+        System.out.println("______________________________");
```
I see that you use the divider quite often in your code. You might consider adding a method which prints the divider as it would save you time from copy and pasting the methods:

```
private void printDivider() {
        System.out.println("______________________________");
}
```

RadieonAjax commented on src/main/java/bean/Bean.java:
```diff
@@ -0,0 +1,143 @@
+
+public class Bean {
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+    private Parser parser;
```
Great job with the clear and concise names!

RadieonAjax reviewed (commented):
Overall, your code is clean and readable, and the header comments makes it really easy to understand what the method is meant to do. Other than a few inconsistencies here and there, your code is superb! Keep up the great work!

snowstopxt commented on src/main/java/bean/parser/Parser.java:
```diff
@@ -0,0 +1,45 @@
+        String details = parts.length > 1 ? parts[1] : "";
+
+        switch (command) {
+            case "list":
```
Does this follow the switch case indentation?

snowstopxt commented on src/main/java/bean/task/TodoTask.java:
```diff
@@ -0,0 +1,47 @@
+     *
+     * @param name The name of the todo task.
+     */
+    public TodoTask(String name) {
```
I like how each class is named with Task at the end to show that they inherit from Task!

snowstopxt commented on src/main/java/bean/task/Task.java:
```diff
@@ -0,0 +1,84 @@
+    /**
+     * Marks the task as completed and updates the icon.
+     */
+    public void completeTask() {
```
Will using the name setComplete show that it is a setter function more explictly?

snowstopxt commented on src/main/java/bean/exception/EmptyDescriptionException.java:
```diff
@@ -0,0 +1,11 @@
+package bean.exception;
+
+public class EmptyDescriptionException extends Exception {
```
I like how you named the specific exception classes

snowstopxt commented on src/main/java/bean/TaskList.java:
```diff
@@ -0,0 +1,61 @@
+        tasks.get(taskIndex).undoTask();
+    }
+
+    public int size() {
```
Will getSize() show that it is a getter function more explictly?

snowstopxt reviewed (commented):
LGTM. A few nits to fix. Great job!

ttzuche committed:
Change colour theme for main window, create gui package

ttzuche committed:
Fix switch case indentation in Storage and Parser

ttzuche committed:
Merge branch-Level-10 into master

ttzuche committed:
Add assert statements to Bean, TaskList, Storage and Task

ttzuche committed:
Merge pull request #2 from ttzuche/branch-A-Assertions

Merge branch-A-Assertions

ttzuche committed:
Extract methods in Bean and Ui

ttzuche committed:
Merge pull request #3 from ttzuche/branch-A-CodeQuality

Merge branch-A-CodeQuality

ttzuche committed:
Merge branch-A-CodeQuality: Refactor repetitive functions in Bean and Ui classes

The Bean and Ui classes contain functions that are repetitive, leading to code
duplication. This increases the risk of inconsistent changes and makes code
less readable.

This needs to change to improve code quality of the codebase. Refactoring
will allow for easier future updates.

Extract the repetitive functions in Bean and Ui classes.

Refactoring to ensure that the logic remains DRY, which enhances maintainability
and minimise potential bugs.

Let's,
* extract all the methods to handle different type of tasks in Bean
* extract the methods returning matching tasks or tasks in general

ttzuche committed:
Merge branch 'master' of https://github.com/ttzuche/ip

ttzuche committed:
Add task edit function

The application lacks the ability to edit existing tasks, which limits
users flexibility in managing their task list.

Implement parsing for the edit command in Parser class.
Handle task edits of different task types.
Add function for handleEditTask in Bean class.

This allows users to efficiently modify task details directly.

ttzuche committed:
Merge pull request #4 from ttzuche/branch-BCD-Extension

Add C-Update function

ttzuche committed:
Add README.md guide to Bean

ttzuche committed:
Add Ui.png and title to the app

ttzuche committed:
Fix sentence structure in README

ttzuche committed:
Fix file path for Ui.png in README

ttzuche committed:
Change file path of Ui.png to relative in README

ttzuche committed:
Fix Javadoc overview phrasing in Launcher class

ttzuche committed:
Place Ui.png in docs directory

ttzuche committed:
Reupload Ui.png to show title
